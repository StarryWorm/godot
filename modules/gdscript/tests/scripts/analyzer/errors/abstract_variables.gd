@abstract class BaseAbstractVar extends Node:
	@abstract var a: int
	@abstract var b: Node2D
	@abstract var c: Vector2

class ConcreteFulfilled extends BaseAbstractVar:
	@override var a = 1
	@override var b = Node2D.new()
	@override var c = Vector2i.ZERO

class ConcreteUnfulfilled extends BaseAbstractVar:
	pass

class ConcreteDeclaresAbstractVar:
	@abstract var d: String

@abstract class DoubleAbstractVar:
	@abstract @abstract var a: int

@abstract class InitializedAbstractVar:
	@abstract var a: int = 1

func test():
	pass
