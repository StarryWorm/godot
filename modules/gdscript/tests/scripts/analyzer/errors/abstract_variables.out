GDTEST_ANALYZER_ERROR
>> ERROR at line 18: "@abstract" annotation can only be used once per variable.
>> ERROR at line 21: An abstract variable cannot have an initializer.
>> ERROR at line 11: Class "ConcreteUnfulfilled" must provide a value for "BaseAbstractVar.a" and other inherited abstract variables or be marked as "@abstract".
>> ERROR at line 14: Class "ConcreteDeclaresAbstractVar" is not abstract but contains abstract variables. Mark the class as "@abstract" or remove "@abstract" from all variables in this class.
